policy/release/labels.rego

```diff
@@ -0,0 +1,155 @@
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	found_labels := {name |
+		some label in labels
+		name := label.name
+	}
+	some required_label in required_labels
+	name := required_label.name
+	not name in found_labels
+	description := required_label.description
+	result := lib.result_helper_with_term(rego.metadata.chain(), [name, description], name)
+}
+
+# METADATA
+# title: Optional labels
+# description: >-
+#   Check the image for the presence of labels that are recommended,
+#   but not required. Use the rule data "optional_labels" key to set
+#   the list of labels to check, or the "fbc_optional_labels" key for
+#   fbc images.
+# custom:
+#   short_name: optional_labels
+#   failure_msg: 'The optional %q label is missing. Label description: %s'
+#   solution: >-
+#     Update the image build process to set the optional labels.
+#   collections:
+#   - redhat
+#
+warn contains result if {
+	found_labels := {name |
+		some label in labels
+		name := label.name
+	}
+	some optional_label in optional_labels
+	name := optional_label.name
+	not name in found_labels
+	description := optional_label.description
+	result := lib.result_helper_with_term(rego.metadata.chain(), [name, description], name)
+}
+
+# METADATA
+# title: Inherited labels
+# description: >-
+#   Check that certain labels on the image have different values than the labels
+#   from the parent image. If the label is inherited from the parent image but not
+#   redefined for the image, it will contain an incorrect value for the image.
+#   Use the rule data "inherited_labels" key to set the list of labels
+#   to check, or the "fbc_inherited_labels" key for fbc images.
+# custom:
+#   short_name: inherited_labels
+#   failure_msg: The %q label should not be inherited from the parent image
+#   solution: >-
+#     Update the image build process to overwrite the inherited labels.
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	some inherited_label in inherited_labels
+	name := inherited_label.name
+	_value(labels, name) == _value(parent_labels, name)
+	result := lib.result_helper_with_term(rego.metadata.chain(), [name], name)
+}
+
+labels contains label if {
+	some name, value in input.image.config.Labels
+	count(value) > 0
+	label := {"name": name, "value": value}
+}
+
+parent_labels contains label if {
+	some name, value in input.image.parent.config.Labels
+	count(value) > 0
+	label := {"name": name, "value": value}
+}
+
+_value(labels, name) := value if {
+	value := [v |
+		some label in labels
+		label.name == name
+		v := label.value
+	][0]
+}
+
+required_labels := lib.rule_data("required_labels") if {
+	not is_fbc
+} else := lib.rule_data("fbc_required_labels")
+
+optional_labels := lib.rule_data("optional_labels") if {
+	not is_fbc
+} else := lib.rule_data("fbc_optional_labels")
+
+inherited_labels := lib.rule_data("inherited_labels") if {
+	not is_fbc
+} else := lib.rule_data("fbc_inherited_labels")
```

Comment:
This pattern seems like it might explode easily. As an example, once we add bundle images, they might have different rules. Are you okay with this precedence?

Reply:
I think it's ok for now. We can revisit it when we do add other image types.

Reply:
We'll likely be making changes to the labels we choose to care about when we're closer to shipping containers that users can actually directly interact with. https://issues.redhat.com/browse/RHTAP-409 

A first pass to help our services get to production makes sense to me, and we can revise from there as needed.